### PR TITLE
rework auto test cases

### PR DESCRIPTION
**Pull request by songbell**
### Details:
 - 1) leverage template for shared tests instantiation, do not depend on cpu/gpu plugin any more
 - 2) leverage mock plugin in auto func tests for supplement of tests not covered in 1), like multiple devices tests, threading of multiple device etc
 
### Tickets:
 - 110083

### CODE REVIEW

**Review by wangleis — APPROVED** [review]

**Review by yangwang201911 — APPROVED** [review]

File: .ci/azure/linux.yml [hunk 1]
```diff
@@ -411,6 +411,9 @@ jobs:
   - script: $(RUN_PREFIX) $(INSTALL_TEST_DIR)/ov_auto_unit_tests --gtest_output=xml:$(INSTALL_TEST_DIR)/TEST-ov_auto_unit_tests.xml
     displayName: 'AUTO UT'
 
+  - script: $(RUN_PREFIX) $(INSTALL_TEST_DIR)/ov_auto_func_tests --gtest_output=xml:$(INSTALL_TEST_DIR)/TEST-ov_auto_func_tests.xml
+    displayName: 'AUTO FuncTests'
+
```
Comment (ilya-lavrenov) [inline .ci/azure/linux.yml:416]: Please, add similar lines to files in .github folder (Windows, Linux pipelines)

File: src/plugins/auto/tests/functional/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+# SPDX-License-Identifier: Apache-2.0
+#
+
+set(TARGET_NAME ov_auto_func_tests)
+
+set(DEPENDENCIES
+    mock_engine
+    openvino::runtime
+    func_test_utils
+    openvino_auto_plugin
```
Comment (ilya-lavrenov) [inline src/plugins/auto/tests/functional/CMakeLists.txt:11]: `DEPENDENCIES` are plugins only, usual libraries are passed as `LINK_LIBRARIES`

File: src/plugins/auto/tests/unit/life_time_test.cpp [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    auto states = request->query_state();
+    auto res_so = mock_states.front()._so;
+    for (auto& state : states)
+        ASSERT_EQ(state._so, res_so);
+}
+
+TEST_F(AutoLifeTimeTest, loaded_tensor_multi) {
+    plugin->set_device_name("MULTI");
+    // get Parameter
+    config.insert(ov::device::priorities("GPU.0"));
```
Comment (ilya-lavrenov) [inline src/plugins/auto/tests/unit/life_time_test.cpp:62]: is it OK that GPU.0 is here?
In the description it's stated the tests don't depend on GPU. Or is it a dummy GPU device?

File: .ci/azure/linux.yml (hunk 1 above)
Comment (songbell) [reply .ci/azure/linux.yml:416]: do you mean .github/workflows/linux.yml and windows.yml? I have already added.

File: src/plugins/auto/tests/unit/life_time_test.cpp (hunk 3 above)
Comment (songbell) [reply src/plugins/auto/tests/unit/life_time_test.cpp:62]: yes, all these used are dummy devices. especially here, the icore is mocked, and the available devices returned from icore are dummy devices we created.

File: src/plugins/auto/tests/functional/CMakeLists.txt (hunk 2 above)
Comment (ilya-lavrenov) [reply src/plugins/auto/tests/functional/CMakeLists.txt:11]: we can merge and fix comment later in case no more comments

**Review by ilya-lavrenov — APPROVED** [review]

**Comment by wenjiew** [comment]
@songbell Can we make this CI green and merge this? Thanks!

File: src/plugins/auto/tests/functional/CMakeLists.txt (hunk 2 above)
Comment (songbell) [reply src/plugins/auto/tests/functional/CMakeLists.txt]: fixed

**Pull request merged**